packages/synapse-react-client/src/synapse-queries/subscription/useSubscription.ts

```diff
@@ -83,32 +83,25 @@ export function useGetAllSubscriptions(
     queryKeyOverride ?? keyFactory.getAllSubscriptionsQueryKey(query),
     async context => {
       const offset = context.pageParam as number | undefined
-      return await SynapseClient.getAllSubscriptions(
+      const subscriptions = await SynapseClient.getAllSubscriptions(
         accessToken,
         10,
         offset,
         query,
       )
+      subscriptions.results.forEach(subscription => {
+        queryClient.setQueryData(
+          keyFactory.getSubscriptionQueryKey(
+            subscription.objectId,
+            subscription.objectType,
+          ),
+          subscription,
+        )
+      })
+      return subscriptions
```

Comment:
Add individual items to the query cache in the `queryFn` instead of in `onSuccess`, which was removed in @tanstack/react-query v5.